scripts/o3de/o3de/engine_template.py

```diff
@@ -181,18 +205,25 @@ def _execute_template_json(json_data: dict,
     # regular copy if not templated
     for copy_file in json_data['copyFiles']:
         # construct the input file name
-        in_file = template_path / 'Template' /copy_file['file']
-
-        # the file can be marked as optional, if it is and it does not exist skip
-        if copy_file['isOptional'] and copy_file['isOptional'] == 'true':
-            if not os.path.isfile(in_file):
-                continue
+        in_file = template_path / 'Template' / copy_file['file']
 
         # construct the output file name
         out_file = destination_path / copy_file['file']
 
         # transform the output file name
         out_file = _transform(out_file.as_posix(), replacements, keep_license_text)
+        out_file = pathlib.Path(out_file)
+
+        if not keep_restricted_in_instance and 'Platform' in out_file.parts:
+            try:
+                # the name of the Platform should follow the '/Platform/'
+                pattern = r'/Platform/(?P<Platform>[^/:*?\"<>|\r\n]+/?)'
+                found_platform = re.search(pattern, out_file.as_posix()).group('Platform')
+                found_platform = found_platform.replace('/', '')
```

Comment:
Instead of using a regular expression, a better approach is to iterate over the [pathlib.parts](https://docs.python.org/3/library/pathlib.html?highlight=pathlib#pathlib.PurePath.part) to locate the "Platform" directory, and then just take the next part element.

This takes the guess work out of trying to parse a path using strings.

```suggestion
                platform_index = out_file.parts.index('Platform')
                if platform_index != -1 and platform_index + 1 < len(out_file.parts):
                    found_platform = out_file.parts[platform_index + 1]
```

Reply:
This logic can just be made into a function instead since it is invoked at least four times throughout

```suggestion
def extract_platform_name(input_path : pathlib.Path) -> str or None:
     platform_name = None
     platform_index = input_path.parts.index('Platform')
     if platform_index != -1 and platform_index + 1 < len(input_path.parts):
         platform_name = input_path.parts[platform_index + 1]
     return platform_name
```